libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -1224,6 +1224,85 @@ def DetachServiceAccount(self) -> None:
       raise errors.ServiceAccountRemovalError('Service account detatchment '
           'failure: {0:s}'.format(str(exception)), __name__)
 
+  def _NormaliseFirewallRules(self, nic_rules: Dict[str, Any]) -> List[Any]:
+    """Normalise firewall policies and firewall rules into a common format.
+
+    Args:
+      nic_rules: the effective firewall rules for an individual NIC.
+
+    Returns:
+      List[Dict[str, Any]]: The normalised firewall rules for a NIC with
+        individual rules in the following format:
+        {
+          'type': 'policy' or 'firewall',
+          'policy_level': int,
+          'priority': int,
+          'direction': 'INGRESS' or 'EGRESS',
+          'l4config': [
+            {
+              'ipProtocol': 'tcp' or 'udp or 'icmp',
+              'ports': List[str]
+            }]
+          'ips': List[str],
+          'action': 'allow' or 'deny' or 'goto_next'
+        }
+    """
+
+    normalised_rules = []
+    firewall_policies = nic_rules['firewallPolicys']
+    firewalls = nic_rules['firewalls']
+
+    for policy_level, policy in enumerate(firewall_policies):
+      for rule in policy['rules']:
+        is_ingress = rule['direction'] == 'INGRESS'
+        normalised_rule = {
+            'type': 'policy',
+            'policy_level': policy_level,
+            'priority': rule['priority'],
+            'direction': rule['direction'],
+            'l4config': rule['match']['layer4Configs'],
+            'ips': (rule['match']['srcIpRanges'] if is_ingress else
+                    rule['match']['destIpRanges']),
+            'action': rule['action']}
+        normalised_rules.append(normalised_rule)
+
+    for rule in firewalls:
+      is_ingress = rule['direction'] == 'INGRESS'
+      is_allow = 'allowed' in rule
+      normalised_rule = {
+          'type': 'firewall',
+          'policy_level': None,
+          'priority': rule['priority'],
+          'direction': rule['direction'],
+          'l4config': rule['allowed'] if is_allow else rule['denied'],
+          'ips': (rule['sourceRanges'] if is_ingress else
+              rule['destinationRanges']),
+          'action': 'allow' if is_allow else 'deny'}
+      normalised_rules.append(normalised_rule)
+
+    return normalised_rules
+
+  def GetEffectiveFirewallRules(self) -> Dict[str, List[Any]]:
+    """Get the effective firewall rules for an instance.
+
+    Returns:
+      Dict[str, List[Any]]: The effective firewall rules per NIC.
+    """
+    gce_instance_client = self.GceApi().instances()
+    instance_info = self.GetOperation()
+    fw_rules = {}
+
+    for nic in instance_info.get('networkInterfaces', []):
+      nic_name = nic['name']
+      request = gce_instance_client.getEffectiveFirewalls(
+          project=self.project_id, instance=self.name, zone=self.zone,
+          networkInterface=nic_name)
+      response = request.execute()
```

Comment:
You may wanna wrap this using `common.ExecuteRequest`, so that errors are caught.